core/src/main/java/ai/timefold/solver/core/impl/score/director/ValueRangeManager.java

```diff
@@ -85,8 +91,10 @@ public static <Solution_> ValueRangeManager<Solution_> of(SolutionDescriptor<Sol
      */
     public ValueRangeManager(SolutionDescriptor<Solution_> solutionDescriptor) {
```

Comment:
All these arrays are based on value range descriptors.
Which means this class is effectively N instances within 1 class; 1 for each value range descriptor.

Arguably, if you refactor the logic so that each VRM works with exactly one VRD, the code will also get much simpler. (You will just have to have one "router" class to delegate requests to these instances based on the specific VRD used.) Something like `valueRangeManager.getFor(valueRangeDescriptor)`.